website/docs/r/nsxt_nat_rule.html.markdown

```diff
@@ -0,0 +1,188 @@
+  snat_destination_addresses = "8.8.8.8"
+  logging = true
+}
+```
+
+## Example Usage 2 (Prevent SNAT for internal addresses in subnet 11.11.11.0/24)
+```hcl
+resource "vcd_nsxt_nat_rule" "no-snat" {
+  org  = "dainius"
+  vdc  = "nsxt-vdc-dainius"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  name        = "test-no-snat-rule"
+  rule_type   = "NO_SNAT"
+  description = "description"
+
+  internal_addresses  = "11.11.11.0/24"
+}
+```
+
+## Example Usage 3 (DNAT rule translates Edge Gateway primary IP to internal IP 11.11.11.2)
+```hcl
+resource "vcd_nsxt_nat_rule" "dnat" {
+  org = "my-org"
+  vdc = "nsxt-vdc"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  name        = "test-dnat-rule"
+  rule_type   = "DNAT"
+  description = "description"
+
+  # Using primary_ip from edge gateway
+  external_addresses = tolist(data.vcd_nsxt_edgegateway.existing.subnet)[0].primary_ip
+  internal_addresses = "11.11.11.2"
+  logging            = true
+}
+```
+
+## Example Usage 4 (No DNAT rule)
+```hcl
+resource "vcd_nsxt_nat_rule" "no-dnat" {
+  org = "my-org"
+  vdc = "nsxt-vdc"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  name      = "test-no-dnat-rule"
+  rule_type = "NO_DNAT"
+
+
+  # Using primary_ip from edge gateway
+  external_addresses = tolist(data.vcd_nsxt_edgegateway.existing.subnet)[0].primary_ip
+  dnat_external_port = 7777
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful
+  when connected as sysadmin working across different organisations.
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level.
+* `edge_gateway_id` - (Required) The ID of the edge gateway (NSX-T only). Can be looked up using
+  `vcd_nsxt_edgegateway` data source
+* `name` - (Required) A name for NAT rule
+* `description` - (Optional) An optional description of the NAT rule
+* `enabled` (Optional) - Enables or disables NAT rule (default `true`)
+* `rule_type` - (Required) One of `DNAT`, `NO_DNAT`, `SNAT`, `NO_SNAT`
+  * `DNAT` rule translates the IP address and optionally the port (using `dnat_external_port`) of
+    packets received by an organization VDC network that are coming from an external network or from
+    another organization VDC network.
+  * `NO_DNAT` rule prevents the translation of the external IP address of packets received by an
+    organization VDC from an external network or from another organization VDC network.
+  * `SNAT` rule translates the source IP address of packets sent from an organization VDC network
+    out to an external network or to another organization VDC network.
+  * `NO_SNAT` rule prevents the translation of the internal IP address of packets sent from an
+    organization VDC out to an external network or to another organization VDC network.
+
+* `external_addresses` (Optional) IP address or CIDR of external network.
```

Comment:
My main concern of this PR. I have noticed inconsistency in using plural vs. singular notions for addresses:
- The fields `external_addresses`, `internal_addresses` and `snat_destination_addresses` are plural.
- Documentation comments for them are singular use "IP address".
- All the tests and docs are providing only a single IP address.

Reply:
Yes. After reviewing UI and docs changing all to be singular `external_address`, `internal_address` and `snat_destination_address`